src/frontend/src/CustomNodes/GenericNode/components/NodeOutputfield/index.tsx

```diff
@@ -0,0 +1,254 @@
+          data.node?.outputs![index].types[0] ??
+          title
+        }
+        nodeId={data.id}
+        frozen={data.node?.frozen}
+        name={title ?? type}
+      />
+    );
+  }
+  const handleUpdateOutputHide = (value?: boolean) => {
+    setNode(data.id, (oldNode) => {
+      let newNode = cloneDeep(oldNode);
+      newNode.data = {
+        ...newNode.data,
+        node: {
+          ...newNode.data.node,
+          outputs: newNode.data.node.outputs?.map((output, i) => {
+            if (i === index) {
+              output.hidden = value ?? !output.hidden;
+            }
+            return output;
+          }),
+        },
+      };
+      return newNode;
+    });
+    updateNodeInternals(data.id);
+  };
+
+  useEffect(() => {
+    if (disabledOutput && data.node?.outputs![index].hidden) {
+      handleUpdateOutputHide(false);
+    }
+  }, [disabledOutput]);
+
+  return !showNode ? (
+    <HandleRenderComponent
+      left={false}
+      nodes={nodes}
+      tooltipTitle={tooltipTitle}
+      id={id}
+      title={title}
+      edges={edges}
+      myData={myData}
+      colors={colors}
+      setFilterEdge={setFilterEdge}
+      showNode={showNode}
+      testIdComplement={`${data?.type?.toLowerCase()}-noshownode`}
+    />
+  ) : (
+    <div
+      ref={ref}
+      className="relative mt-1 flex w-full flex-wrap items-center justify-between bg-muted px-5 py-2"
+    >
+      <>
+        <div className="flex w-full items-center justify-end truncate text-sm">
+          <div className="flex-1">
+            <Button
+              disabled={disabledOutput}
+              unstyled
+              onClick={() => handleUpdateOutputHide()}
+              data-testid={`input-inspection-${title.toLowerCase()}`}
+            >
+              <IconComponent
+                className={cn(
+                  "h-4 w-4",
+                  disabledOutput ? "text-muted-foreground" : "",
+                )}
+                strokeWidth={1.5}
+                name={data.node?.outputs![index].hidden ? "EyeOff" : "Eye"}
+              />
+            </Button>
+          </div>
+          <Case condition={data.node?.frozen}>
+            <div className="pr-1">
+              <IconComponent className="h-5 w-5 text-ice" name={"Snowflake"} />
+            </div>
+          </Case>
+
+          <div className="flex gap-2">
+            <span className={data.node?.frozen ? "text-ice" : ""}>
+              {renderTitle()}
+            </span>
+            <ShadTooltip
+              content={
+                displayOutputPreview
+                  ? unknownOutput
+                    ? "Output can't be displayed"
+                    : "Inspect Output"
+                  : "Please build the component first"
+              }
+            >
+              <Button
+                unstyled
+                disabled={!displayOutputPreview || unknownOutput}
+                onClick={() => setOpenOutputModal(true)}
```

Comment:
maybe instead of setting the output modal open here we can put the output modal as a wrapper, so that the button can be the trigger

Reply:
if we do it that way we will render the modal even when it's close, I don't think we want that because all the data we use inside this modal does not exist yet, so render it before is not good IMO

Reply:
we should implement nullable checks instead of that